src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -6109,6 +6110,30 @@ def update_run(
             session.commit()
             session.refresh(existing_run)
 
+            # Add logs if specified
+            if run_update.add_logs:
+                for log_request in run_update.add_logs:
+                    # Validate the artifact store exists
+                    self._get_reference_schema_by_id(
+                        resource=log_request,
+                        reference_schema=StackComponentSchema,
+                        reference_id=log_request.artifact_store_id,
+                        session=session,
+                        reference_type="logs artifact store",
+                    )
+
+                    # Create the log entry
+                    log_entry = LogsSchema(
+                        uri=log_request.uri,
+                        title=log_request.title,
+                        pipeline_run_id=existing_run.id,
+                        artifact_store_id=log_request.artifact_store_id,
+                    )
+                    session.add(log_entry)
+
+                session.commit()
+                session.refresh(existing_run)
```

Comment:
These refresh calls are DB calls each time, and are not necessary in this case I think. Can you try without it and see if it still shows up in the response of the update endpoint?